### PR TITLE
Meta search from conf

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by trasher** [comment]
logDebug output:

```
Linking state_types  {"user":"2@johanxps"} 
Linking document_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking report_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking infocom_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking reservation_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking location_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking ticket_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking link_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.28Mio)"} 
Linking networkport_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
Linking contract_types  {"user":"2@johanxps","mem_usage":"0.001\", 9.29Mio)"} 
Linking certificate_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
Linking cluster_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
Linking operatingsystem_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
Linking domain_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
Linking appliance_types  {"user":"2@johanxps","mem_usage":"0.000\", 9.29Mio)"} 
```

File: inc/search.class.php [hunk 1]
```diff
@@ -2187,14 +2187,36 @@ static function getMetaItemtypeAvailable($itemtype) {
...
+      global $CFG_GLPI;
+      $excludeds = [
+         'report',
+         'infocom',
+         'link',
+         'networkport',
+         'reservation'
+      ];
+      foreach ($CFG_GLPI as $key => $values) {
+          $matches = [];
```
Comment (cedric-anne) [inline inc/search.class.php:2199]:
```suggestion
         $matches = [];
```

File: inc/search.class.php [hunk 2]
```diff
@@ -2187,14 +2187,36 @@ static function getMetaItemtypeAvailable($itemtype) {
...
+         if (preg_match('/^(.+)_types$/', $key, $matches)) {
+             $itemtype = $matches[1];
```
Comment (cedric-anne) [inline inc/search.class.php:2201]:
```suggestion
            $itemtype = $matches[1];
```

File: inc/search.class.php [hunk 3]
```diff
@@ -2187,14 +2187,36 @@ static function getMetaItemtypeAvailable($itemtype) {
...
+            if (in_array($itemtype, $excludeds)) {
+               continue;
+            }
+             $class = ucwords($itemtype);
```
Comment (cedric-anne) [inline inc/search.class.php:2205]:
```suggestion
            $class = ucwords($itemtype);
```

**Review by cedric-anne — APPROVED** [review]

File: inc/search.class.php [hunk 4]
```diff
@@ -5202,13 +5225,24 @@ static function addMetaLeftJoin($from_type, $to_type, array &$already_link_table
...
+         default:
+            $cfg_type = strtolower($to_type) . '_types';
+            if (isset($CFG_GLPI[$cfg_type])) {
+               $items_class = $to_type . '_Item';
+               if (!class_exists($items_class)) {
+                  $items_class = 'Item_' . $to_type;
+               }
+
+               if (class_exists($items_class)) {
+                  array_push($already_link_tables2, $to_table);
```
Comment (cedric-anne) [inline inc/search.class.php:5237]:
```suggestion
                  array_push($already_link_tables2, $items_table);
                  array_push($already_link_tables2, $to_table);
```

File: inc/search.class.php [hunk 5]
```diff
@@ -5200,13 +5223,25 @@ static function addMetaLeftJoin($from_type, $to_type, array &$already_link_table
...
+            if (isset($CFG_GLPI[$cfg_type])) {
+               $items_class = $to_type . '_Item';
+               if (!class_exists($items_class)) {
+                  $items_class = 'Item_' . $to_type;
+               }
+
+               if (class_exists($items_class)) {
+                  array_push($already_link_tables2, $items_table);
+                  array_push($already_link_tables2, $to_table);
+                  $items_table = $items_class::getTable();
```
Comment (cedric-anne) [inline inc/search.class.php:5237]:
```
==> Error E_WARNING in /var/glpi/tests/functionnal/Search.php on line 536, generated by file /var/glpi/inc/search.class.php on line 5235:
Undefined variable $items_table
```

```suggestion
                  $items_table = $items_class::getTable();
                  array_push($already_link_tables2, $items_table);
                  array_push($already_link_tables2, $to_table);
```

**Pull request closed without merging**